functions/src/role.ts

```diff
@@ -0,0 +1,50 @@
+import type { Lobby } from "./firestore-types/lobby";
+import { getPrivatePlayerCollection } from "./firestore-collections";
+import { db } from "./app";
+import { firestore } from "firebase-admin";
+
+export function assignRole(lobby: firestore.DocumentReference<Lobby>) {
```

Comment:
I had some ideas for how to improve this so I created #115

Reply:
pass in the transaction as a parameter

Reply:
pass in DocumentSnapshot instead of Document Reference